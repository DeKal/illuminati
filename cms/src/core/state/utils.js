export const toSuccess = actionName => `${actionName}Success`

export const toError = actionName => `${actionName}Error`

export const toRequest = actionName => `${actionName}Request`
