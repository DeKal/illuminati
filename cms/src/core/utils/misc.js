import { POSTS_PER_PAGE } from 'core/const/posts'

export const isPathMatchUrl = (pathName, url) => {
  if (url === '/' || pathName === '/') {
    return pathName === url
  }
  return pathName.includes(url)
}

export const stringtifyTags = (tags, limits = 20) => {
  const tagsStr = tags.join(', ')
  if (tagsStr.length > limits) {
    return tagsStr.slice(0, limits) + '...'
  }
  return tagsStr
}

export const calcMaxRowsPerPage = postCount => {
  const postsPerPageOptions = []
  for (let index = 0; index < POSTS_PER_PAGE.length; index++) {
    postsPerPageOptions.push(POSTS_PER_PAGE[index])
    if (POSTS_PER_PAGE[index] > postCount) {
      break
    }
  }
  return postsPerPageOptions
}
