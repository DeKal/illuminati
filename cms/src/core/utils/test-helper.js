import React from 'react'
import { shallow, mount } from 'enzyme'
import Theme from 'core/theme/ThemeWrapper'

export const shallowWithTheme = component => {
  const wrapper = shallow(
    <Theme>
      <div data-test-id="theme-wrapper">{component}</div>
    </Theme>
  )
  return wrapper.find('[data-test-id="theme-wrapper"]').shallow()
}

export const mountWithTheme = component => {
  const wrapper = mount(
    <Theme>
      <div data-test-id="theme-wrapper">{component}</div>
    </Theme>
  )
  return wrapper.find('[data-test-id="theme-wrapper"]')
}
