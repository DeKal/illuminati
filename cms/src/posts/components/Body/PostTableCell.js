import { withStyles } from '@material-ui/core/styles'
import TableCell from '@material-ui/core/TableCell'

const PostTableCell = withStyles(theme => ({
  root: {
    width: '50%',
    cursor: 'pointer',
    whiteSpace: 'nowrap',
    '&:hover': {
      color: theme.colors.primary
    }
  }
}))(TableCell)

export default PostTableCell
