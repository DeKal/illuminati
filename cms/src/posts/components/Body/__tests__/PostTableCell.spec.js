import * as React from 'react'
import { mountWithTheme } from 'core/utils/test-helper'
import PostTableCell from '../PostTableCell'

describe('PostTableCell', () => {
  it('should render with no error', function() {
    const wrap = mountWithTheme(<PostTableCell />)
    expect(wrap).toMatchSnapshot()
  })
})
