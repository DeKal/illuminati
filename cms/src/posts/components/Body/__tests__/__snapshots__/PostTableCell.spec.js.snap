// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostTableCell should render with no error 1`] = `
<div
  data-test-id="theme-wrapper"
>
  <WithStyles(WithStyles(ForwardRef(TableCell)))>
    <WithStyles(ForwardRef(TableCell))
      classes={
        Object {
          "root": "WithStyles(ForwardRef(TableCell))-root-1",
        }
      }
    >
      <ForwardRef(TableCell)
        classes={
          Object {
            "alignCenter": "MuiTableCell-alignCenter",
            "alignJustify": "MuiTableCell-alignJustify",
            "alignLeft": "MuiTableCell-alignLeft",
            "alignRight": "MuiTableCell-alignRight",
            "body": "MuiTableCell-body",
            "footer": "MuiTableCell-footer",
            "head": "MuiTableCell-head",
            "paddingCheckbox": "MuiTableCell-paddingCheckbox",
            "paddingNone": "MuiTableCell-paddingNone",
            "root": "MuiTableCell-root WithStyles(ForwardRef(TableCell))-root-1",
            "sizeSmall": "MuiTableCell-sizeSmall",
            "stickyHeader": "MuiTableCell-stickyHeader",
          }
        }
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root WithStyles(ForwardRef(TableCell))-root-1"
        />
      </ForwardRef(TableCell)>
    </WithStyles(ForwardRef(TableCell))>
  </WithStyles(WithStyles(ForwardRef(TableCell)))>
</div>
`;
